Adding a section on parallel assignment

rrosenblum opened the pull request:
As discussed in the RuboCop pull request for parallel assignment, https://github.com/bbatsov/rubocop/pull/1875.

I wasn't sure where exactly to put this. Let me know if it needs tweaking or moving.

bbatsov commented:
A small remark about the commit message - Adding -> Add.

bbatsov commented on README.md:
````diff
@@ -635,6 +635,43 @@ Translations of the guide are available in the following languages:
    ```
 
+* <a name="parallel-assignment"></a>
+    Avoid the use of parallel assignment for defining variables, except for
````
Maybe you can break down this long sentence into two sentences?

bbatsov commented on README.md:
```diff
@@ -635,6 +635,43 @@ Translations of the guide are available in the following languages:
+* <a name="parallel-assignment"></a>
+    Avoid the use of parallel assignment for defining variables, except for
+    the return of method calls, use with the splat operator, or variable
+    reassignment. Parallel assignment is less readable and slower.
```
`and a bit slower` - let's not put heavy emphasis on this

bbatsov commented on README.md:
```diff
@@ -635,6 +635,43 @@ Translations of the guide are available in the following languages:
+  c = 'baz'
+  d = 'foobar'
+
+  # good - variable reassignment
```
Maybe we should go into a bit more details why this particular reassignment is good (as there are other cases which won't be - e.g. `a, b = 1, 2; a, b = 3, 4`.

rrosenblum replied:
I agree, having some extra information about variable reassignment would be good. I added a sub section on variable reassignment. I think that it may need to be rewritten though.

bbatsov commented on README.md:
```diff
@@ -635,6 +635,47 @@ Translations of the guide are available in the following languages:
+    assignment is less readable than separate assignment. It is also slightly
+    slower than separate assignment.
+<sup>[[link](#parallel-assignment)]</sup>
+ *  Variable reasignment is acceptable because it allows you to swap values
```
I'd just move this into the code examples as a longer comment.

rrosenblum commented:
I have changed the naming of variable reassignment to variable swapping. I think that this better defines what is being done (I plan on making the same naming change inside of RuboCop). I have also moved the explanation to the comments section. Let me know what you think of the changes.

bbatsov commented:
:+1: Looks good. Thanks!
